lib/sequent/core/persistors/replay_optimized_postgres_persistor.rb

```diff
@@ -287,22 +287,18 @@ def do_with_record(record_class, where_clause)
         end
 
         def find_records(record_class, where_clause)
-          if @record_index.use_index?(record_class, where_clause)
-            @record_index.find(record_class, where_clause)
-          else
-            @record_store[record_class].select do |record|
-              where_clause.all? do |k, v|
-                expected_value = v.is_a?(Symbol) ? v.to_s : v
-                actual_value = record[k.to_sym]
-                actual_value = actual_value.to_s if actual_value.is_a? Symbol
-                if expected_value.is_a?(Array)
-                  expected_value.include?(actual_value)
-                else
-                  actual_value == expected_value
-                end
+          (@record_index.find(record_class, where_clause) || @record_store[record_class].select do |record|
```

Comment:
We still need to have `use_index?` here since `find` will always return an empty array if not found, which is `truthy`. We should add a spec for this scenario